### PR TITLE
update German translation

### PR DESCRIPTION
**Pull request by horst-p-w-neubauer**
update German translation

this is done
* 'record' to 'Datensatz'
* 'Constraint' to 'Beschränkungen' in different ways.
* 'clause' to 'Anweisung'
* strict case-issue 'Rowid'
* 'unmittelbar' in stead of 'inline' (context sensitive)
* 'vacuum' to 'Komprimierung'
* 'commit' to 'abschließen'
* don't translate 'Concordance' as it is a trademark
* tried to correct or clearify some missunderstandings

### CODE REVIEW

**Comment by horst-p-w-neubauer** [comment]
Please, take this pullrequest as a draft for an open discussion.
I' have some concerns with some of my suggestions as well.
e.g. 'Prüfbechräunkung' for 'check constraint'

File: src/translations/sqlb_de.ts [hunk 1]
```diff
@@ -5171,15 +5171,15 @@ Grund: %1</translation>
     </message>
     <message>
         <source>Duplicate records</source>
-        <translation type="vanished">Doppelte Einträge</translation>
+        <translation type="vanished">Doppeltee Datensätze</translation>
```
Comment (mgrojo) [inline src/translations/sqlb_de.ts:5174]: Please, note that German is not my native language, but isn't it here a mistake in the second added "e" in "Doppeltee"?

Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:5174]: There is a typo inside here.

File: src/translations/sqlb_de.ts [hunk 2]
```diff
@@ -7132,7 +7132,7 @@ p, li { white-space: pre-wrap; }
     <message>
         <location filename="../SqlUiLexer.cpp" line="67"/>
         <source>() The changes() function returns the number of database rows that were changed or inserted or deleted by the most recently completed INSERT, DELETE, or UPDATE statement.</source>
-        <translation>() Die changes()-Funktion gibt die Anzahl der Datenbankzeilen zurück, die mit dem zuletzt abgeschlossenen INSERT-, DELETE- oder UPDATE-Statement geändert, einfügt oder gelöscht worden sind.</translation>
+        <translation>() Die changes()-Funktion gibt die Anzahl der Datenbankzeilen zurück, die mit dem zuletzt abgeschlossenen INSERT-, DELETE- oder UPDATE-Statements geändert, einfügt oder gelöscht worden sind.</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:7135]: Why plural? Only the last statement is considered here, so I do not see a need for the plural here.

File: src/translations/sqlb_de.ts (hunk 1 above)
Comment (horst-p-w-neubauer) [reply src/translations/sqlb_de.ts:5174]: absolute! Thx

File: src/translations/sqlb_de.ts [hunk 3]
```diff
@@ -108,12 +108,12 @@
     <message>
         <location filename="../AddRecordDialog.ui" line="93"/>
         <source>When you edit the values in the upper frame, the SQL query for inserting this new record is shown here. You can edit manually the query before saving.</source>
-        <translation>Wenn Sie die Werte im oberen Teil ändern, wird hier das SQL-Query für das Einfügen der neuen Zeile angezeigt. Sie können das Query vor dem Speichern manuell bearbeiten.</translation>
+        <translation>Wenn Sie die Werte im oberen Teil ändern, wird hier das SQL-Query für das Einfügen des neuen Datensatz angezeigt. Sie können das Query vor dem Speichern manuell bearbeiten.</translation>
```
[the remark this answers is not in the document]
Comment (horst-p-w-neubauer) [reply src/translations/sqlb_de.ts:111]: yes, thx

File: src/translations/sqlb_de.ts (hunk 2 above)
Comment (horst-p-w-neubauer) [reply src/translations/sqlb_de.ts:7135]: good point, my fault

**Comment by horst-p-w-neubauer** [comment]
> 
> 
> I am not sure whether not translating some of the strings would be the preferred way. Examples:
> 
>     * I would tend to rather use "UNIQUE-Beschränkung" instead of translating "unique" into German, as "UNIQUE" is SQL syntax.
> 
>     * The same for "CHECK-Beschränkung".
> 
>     * The same for "WITHOUT ROWID".
> 
>     * I am not sure about whether "Rowid", "rowid" or "ROWID" would be the best choice.
> 
I agree with "UNIQUE-Beschränkung" and "CHECK-Beschränkung" - great solution!
I disagree with "WITHOUT Rowid" as it is also a dialog checkbox label where it might look like a not translated foreign part.

**Comment by horst-p-w-neubauer** [comment]
Allright. No more commits for today.

**Comment by justinclift** [comment]
Looks like good progress @horst-p-w-neubauer. :smile:

**Comment by horst-p-w-neubauer** [comment]
Now I've checked all items on my todo list.
I suggest to add a manualy updated dictionary as design-document for the translation-process.
I've attached mine as a draft.  I don't know how you use to handle design-documents.
Maybe Excel isn't first choice for erveryone. 
[dictionary_de.xlsx](https://github.com/sqlitebrowser/sqlitebrowser/files/4498785/dictionary_de.xlsx)

**Comment by mgrojo** [comment]
You have also the possibility of using issues or discussions in our Translators team (which you can join if you want). It's easy to write tables or even paste them from spreadsheets. You lose filtering capabilities, but maybe it is easier to share and edit in group. The wiki would be another option for this kind of coordination, but strangely I could not copy the table there. The syntax is the same, though, so it an issue can be used as intermediary for the conversion from spreadsheets to wiki. Another option is using https://dbhub.io/ for storing the dictionary as a DB file. It is then easy to edit it using DB Browser.

Here an example:

Quelle | Kontext | Übersetzung | Bemerkungen
-- | -- | -- | --
bar | Diagramme | Säulen | Säulendiagramme ordnen die Werte von oben nach unten, Balkendiagramme von rechts nach links
bar | toolbar | Werkzeugleiste | siehe: https://community.kde.org/KDE_Localization/de/StandardUebersetzungen
Check-Constraint |   | CHECK-Beschränkung | siehe: https://github.com/sqlitebrowser/sqlitebrowser/pull/2193
clause | SQL | Anweisung |  
Collation |   | Sortierung |  
commit |   | abschließen |  
Constraint |   | Beschränkung | siehe: https://github.com/sqlitebrowser/sqlitebrowser/pull/2193
Log |   | Protokoll | siehe: https://community.kde.org/KDE_Localization/de/StandardUebersetzungen
Message from database [engine]DB says |   | DB meldet | Vorschlag: horst-p-w-neubauer
pattern | regulärer Ausdruck | Suchmuster |  
Query |   | Abfrage |  
Record |   | Datensatz |  
Row |   | Zeile oder Datenbankzeile |  
Statement |   | Anweisung |  
stop |   | abbrechen |  
Toolbar |   | Werkzeugleiste | siehe: https://community.kde.org/KDE_Localization/de/StandardUebersetzungen
Unique-Constraint |   | UNIQUE-Beschränkung | siehe: https://github.com/sqlitebrowser/sqlitebrowser/pull/2193
Vacuum |   | Komprimierung |

**Comment by FriedrichFroebel** [comment]
I have added a new wiki page at https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations to maintain the dictionary there.  We might want to add a general explanation for the translation workflow as well - if I remember correctly, there have been some discussions regarding the required steps some time ago inside an issue.

**Comment by horst-p-w-neubauer** [comment]
Great idea. I love wikis!

**Comment by horst-p-w-neubauer** [comment]
Before this pull-request might get merged I wonder, are there any german documentations that rely on the former translations. As this merge is going to change a lot, I don't want to break anything.

**Comment by mgrojo** [comment]
Awesome!

I've updated the wiki page with the information Martin wrote in https://github.com/sqlitebrowser/sqlitebrowser/issues/182#issuecomment-70413095



> Before this pull-request might get merged I wonder, are there any german documentations that rely on the former translations. As this merge is going to change a lot, I don't want to break anything.

The only documentation that we have is the wiki (which is currently only English, but it could be translated as well) and the "What's This" information, whose strings are already present in the translation files.

**Comment by horst-p-w-neubauer** [comment]
I'm not sure what is the process that comes up with this message:
<location filename="../sqlitedb.cpp" line="1816"/>
        '...  Here's the SQL statement which you might want to fix and execute manually:'

Does it present an artificial produced SQL statement, that might fix the problem or is it the SQL statement that produced the problem and you offer the possibility to fix it and run it manually?

**Comment by horst-p-w-neubauer** [comment]
If there are no more hints I would ask You to commit this request so we can see the translation-result ot the reset shortcuts with the next 3.12 build. BTW - when can we expect to see this?

**Comment by horst-p-w-neubauer** [comment]
If we find this working for german - we should do it for the other langs too.

**Comment by FriedrichFroebel** [comment]
You should probably remove `.project`, `.settings/org.eclipse.core.resources.prefs` and `src/MainWindow.ui.autosave` introduced during your last commit from the PR. Maybe we should add these files/directories/file extensions to the `.gitignore` as well to avoid pushing them again.

**Comment by scottfurry** [comment]
definitely remove IDE-specific files

**Comment by justinclift** [comment]
What's the state of this now?  Is it ready to merge? :smile:

File: .gitignore [hunk 4]
```diff
@@ -38,3 +38,5 @@ libs/*/*/*.a
 
 # Ignore .DS_Store files on OSX
 .DS_Store
+.project
```
Comment (justinclift) [inline .gitignore:41]: These .gitignore changes look like they're not needed?

Comment (FriedrichFroebel) [reply .gitignore:41]: See https://github.com/sqlitebrowser/sqlitebrowser/pull/2193#issuecomment-618263402 - some wrong files have accidentally been committed before. The only part one might argue is that the section used there might not be appropriate.

Comment (justinclift) [reply .gitignore:41]: No worries. :smile:

**Comment by justinclift** [comment]
I'd like to merge this now, if people feel this is ready?

**Comment by horst-p-w-neubauer** [comment]
the conflicted line allready vanished with #2224 
Is there anything, I can do?

**Comment by justinclift** [comment]
I'm ok to merge this, if you feel it's ready?

**Comment by horst-p-w-neubauer** [comment]
I'd say: ready enough to merge.

**Comment by justinclift** [comment]
Awesome, thanks heaps for this too @horst-p-w-neubauer. :smile:

**Pull request merged**